vip_hci/phot/fakecomp.py

```diff
@@ -47,77 +52,168 @@ def inject_fcs_cube(array, psf_template, angle_list, flevel, plsc, rad_dists,
+            raise ValueError(msg)
+
+        sizey = array.shape[2]; sizex = array.shape[3]
+        size_fc = psf_template.shape[2] # considering square frames
+        nframes_wav = array.shape[0]
+        nframes_adi = array.shape[1]
+        fc_fr = np.zeros((nframes_wav, sizey, sizex), dtype=np.float64) # now a 3d cube
+        n_fc_rad = rad_dists.shape[0]
+
+        for i in range(nframes_wav):
+            w = int(np.floor(size_fc/2.))
+            # fcomp in the center of a zeros frame
+            if (psf_template[0].shape[1]%2)==0:
+                fc_fr[i, ceny-w:ceny+w, cenx-w:cenx+w] = psf_template[i]
+            else:
+                fc_fr[i, ceny-w:ceny+w+1, cenx-w:cenx+w+1] = psf_template[i]
+
+        array_out = np.zeros_like(array)
+
+        #if len(theta)>1:
+
+        #else:
+        for fr in range(nframes_adi):
+            tmp = np.zeros_like(fc_fr)
+            for branch in range(n_branches):
+                ang = (branch * 2 * np.pi / n_branches) + np.deg2rad(theta)
+                for i in range(n_fc_rad):
+                    rad = rad_dists[i]
+                    y = rad * np.sin(ang - np.deg2rad(angle_list[fr]))
+                    x = rad * np.cos(ang - np.deg2rad(angle_list[fr]))
+                    if isinstance(flevel,int)==True:
+                        tmp += cube_shift(fc_fr, y, x, imlib=imlib)*flevel
+                    else:
+                        #tmp += cube_shift(fc_fr, y, x, imlib=imlib)*flevel
+                        shift = cube_shift(fc_fr, y, x, imlib=imlib)
+                        #tmp += shift * flevel
+                        tmp += [shift[i]*flevel[i] for i in range(len(flevel))]
+            array_out[:,fr] = array[:,fr] + tmp
+
+        if verbose:
+            if len(theta)>1:
+                for branch in range(n_branches):
+                    print 'Branch '+str(branch+1)+':'
+                    for i in range(n_fc_rad):
+                        ang = (branch * 2 * np.pi / n_branches) + np.deg2rad(theta)
+                        posy = rad_dists[i] * np.sin(ang[i]) + ceny
+                        posx = rad_dists[i] * np.cos(ang[i]) + cenx
+                        rad_arcs = rad_dists[i]*plsc
+                        msg ='\t(X,Y)=({:.2f}, {:.2f}) at {:.2f} arcsec ({:.2f} pxs)'
+                        print msg.format(posx, posy, rad_arcs, rad_dists[i])
+            else:
+                for branch in range(n_branches):
+                    print 'Branch '+str(branch+1)+':'
+                    for i in range(n_fc_rad):
+                        ang = (branch * 2 * np.pi / n_branches) + np.deg2rad(theta)
+                        posy = rad_dists[i] * np.sin(ang) + ceny
+                        posx = rad_dists[i] * np.cos(ang) + cenx
+                        rad_arcs = rad_dists[i]*plsc
+                        msg ='\t(X,Y)=({:.2f}, {:.2f}) at {:.2f} arcsec ({:.2f} pxs)'
+                        #print msg.format(posx, posy, rad_arcs, rad_dists)
```

Comment:
I don't get this if else structure. In the else case you are not printing anything